src/batched/KokkosBatched_Gemm_Decl.hpp

```diff
@@ -358,13 +452,54 @@ int BatchedGemm(const BatchedGemmHandleType *handle, const ScalarType alpha,
                                         Algo::Gemm::Blocked>::type>::type>::
           type;
 
-      // TODO: Add additional heuristic conditionals that look at M,N to
-      // determine
-      //       whether to invoke BatchedSerialGemm, BatchedDblBuf, etc.
-      ret = BatchedSerialGemm<ArgTransA, ArgTransB, bsgModeType, ArgBatchSzDim,
-                              bsgResultsPerThread, ScalarType, AViewType,
-                              BViewType, CViewType>(alpha, A, B, beta, C)
+      // if (on_gpu && c_m >= 20 &&
+      //     (alpha == 1.0F && beta == 0.0F) ? c_m <= 24 : c_m <= 21) {
+      //   // TODO: invoke TeamShmem
+      // } else
+      if (on_gpu &&
+          ((std::is_same<layout_type, Kokkos::LayoutLeft>::value)
+               ? (c_m >= 16)
+               : (c_m >= 24 && c_m <= 32) || (c_m >= 45 && c_m <= 64))) {
+        handle->teamSz = handle->vecLen = 8;
+        // constexpr int tile_m = 32, tile_n = 32, tile_k = 8;
+        constexpr int tile_m = 32, tile_n = 32, tile_k = 8;
+        if (c_m % 32 == 0)  // No bounds checking
+          ret =
+              Impl::BatchedDblBufGemm<ArgTransA, ArgTransB, ArgBatchSzDim,
+                                      BatchedGemmHandleType, ScalarType,
+                                      AViewType, BViewType, CViewType,
+                                      BoundsCheck::No, tile_m, tile_n, tile_k>(
+                  handle, alpha, A, B, beta, C)
+                  .invoke();
+        else
+          ret =
+              Impl::BatchedDblBufGemm<ArgTransA, ArgTransB, ArgBatchSzDim,
+                                      BatchedGemmHandleType, ScalarType,
+                                      AViewType, BViewType, CViewType,
+                                      BoundsCheck::Yes, tile_m, tile_n, tile_k>(handle, alpha, A, B, beta, C)
+                  .invoke();
+      } else {
+        ret = Impl::BatchedSerialGemm<ArgTransA, ArgTransB, bsgModeType,
+                                      ArgBatchSzDim, bsgResultsPerThread,
+                                      ScalarType, AViewType, BViewType,
+                                      CViewType>(alpha, A, B, beta, C)
                 .invoke();
+      }
+      break;
+
+    case GemmKokkosBatchedAlgos::KK_DBLBUF:
+      // Note: The tile sizes of 1x1x1 here will not perform well but must be
+      // selected in order to function on all devices since the serial execution
+      // space has a max team size of 1. KokkosKernels API users will need to
+      // follow an approach similar to KK_SQUARE above for best performance.
+
+      // TODO: Add auto-selection of tile size based on inputs and device type
+      ret =
+          Impl::BatchedDblBufGemm<ArgTransA, ArgTransB, ArgBatchSzDim,
+                                  BatchedGemmHandleType, ScalarType, AViewType,
+                                  BViewType, CViewType, BoundsCheck::Yes, 1, 1,
```

Comment:
Is it possible for users to choose tile size?

Reply:
Yes, but they must use the `Impl::BatchedDblBufGemm` interface.